service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -361,4 +387,40 @@ public PageResponse<List<PipelineRun>> findPipelineRunsPaginated(
   public long getPipelineRunCount(String userId) {
     return pipelineRunsRepository.countByUserId(userId);
   }
+
+  private Sort.Direction validateSortDirection(String sortDirection) {
+    if (sortDirection == null) {
+      return Sort.Direction.DESC;
+    }
+    try {
+      return Sort.Direction.valueOf(sortDirection.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(
+          "Invalid sort direction: %s. Valid values are ASC or DESC".formatted(sortDirection));
+    }
+  }
+
+  private String validateSortProperty(String sortProperty) {
```

Comment:
there's a bit of a quirk here that requires the caller of this to know the internal representation of a PipelineRun. in `getAllPipelineRuns` (v1 and v2), we return a ApiPipelineRun which changes some of the property names:

i.e. created -> timeSubmitted

in order to sort on the field, the caller needs to specify what's stored in the `PipelineRun` table, `created` in this case. but in the response they get, they'll see `timeSubmitted`. so it might be confusing.

i'd love to get thoughts on this. i could add a step to translate the user-facing fields and turn them into a field that actually exists on PipelineRun? It feels a bit messy though...

Reply:
yea im thinking either have the UI handle this or maybe we just keep the returned values the same as the able column names?  Like just return "created" to the user".  Lots of cool parking lots to talk about!

Reply:
it's easy to brush over in the ui since the sort field is manually specified per column.

here's where it'll specify `created` instead of `timeSubmitted` https://github.com/DataBiosphere/terra-ui/pull/5413/files#diff-3affab90042cf5362b6a8608bb35fcf2ed8e25e942086e657a05ee8db7fb1080R188